onnxruntime/core/providers/cuda/tensor/split.cc

```diff
@@ -64,33 +64,35 @@ Status Split::ComputeInternal(OpKernelContext* ctx) const {
     }
   }
 
-  output_ptr.CopyToGpu();
+  if (before_dims > 0 && block_size_inside_axis_dim > 0) {
+    output_ptr.CopyToGpu();
```

Comment:
If this is to fix zero sized input, I would have expected the test to be input_tensor->Shape().Size() == 0. Would that be clearer?